src/main/app/claims/models/response/core/paymentIntention.ts

```diff
@@ -4,22 +4,26 @@ import { PaymentOption } from 'claims/models/response/core/paymentOption'
 import { RepaymentPlan } from 'claims/models/response/core/repaymentPlan'
 import { MomentFactory } from 'shared/momentFactory'
 
-export interface PaymentIntention {
+export class PaymentIntention {
   paymentOption?: PaymentOption
   paymentDate?: Moment
   repaymentPlan?: RepaymentPlan
-}
 
-export namespace PaymentIntention {
-  export function deserialize (input: any): PaymentIntention {
-    return input && {
-      paymentOption: input.paymentOption as PaymentOption,
-      paymentDate: input.paymentDate && MomentFactory.parse(input.paymentDate),
-      repaymentPlan: input.repaymentPlan && {
-        instalmentAmount: input.repaymentPlan.instalmentAmount,
-        firstPaymentDate: MomentFactory.parse(input.repaymentPlan.firstPaymentDate),
-        paymentSchedule: input.repaymentPlan.paymentSchedule
-      }
+  deserialize (input: any): PaymentIntention {
+    if (input) {
```

Comment:
Behaviour is changes as now you return new instance every time even for undefined input. Please revert.

Reply:
I mean please revert behaviour, keep class.